assets/js/components/ModuleRootComponents.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * ModuleRootComponents component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useSelect } from 'googlesitekit-data';
+import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+
+function ModuleRootComponents( { dashboardType } ) {
+	const modules = useSelect( ( select ) =>
+		select( CORE_MODULES ).getModules()
+	);
+
+	const filteredModules = [];
+	if ( modules ) {
+		for ( const moduleSlug in modules ) {
+			if (
+				modules[ moduleSlug ].connected &&
```

Comment:
I'm not sure we want to enforce that these components only render when connected. The AC mentions that in most cases the effects should only run but there could be cases where an effect may be needed when only active, but I'm not sure. (AdSense comes to mind as a possibility)

Reply:
I changed it to `active`. Basically it can go either way, since we do not have many effects, and they are only in GA4 module. We can be a bit flexible as it doesn't make much of difference at the moment, but as you mentioned if we add effects that should run when module is `active` the implementation will cover it as is.